### PR TITLE
Fix incomplete `set_qos` making not working set a default entity qos [13836]

### PR DESCRIPTION
**Pull request by richiware**
There are missing things which not make work well setting a default qos for DataWriters and DataReaders. This PR fixes this and also add/extend a regression test.

Also comes with minor changes to make work Swig when generating Python bindings.

### CODE REVIEW

File: include/fastrtps/utils/collections/ResourceLimitedContainerConfig.hpp [hunk 1]
```diff
@@ -26,6 +26,8 @@
 namespace eprosima {
 namespace fastrtps {
 
+#define dummy_avoid_winmax
```
Comment (MiguelCompany) [inline include/fastrtps/utils/collections/ResourceLimitedContainerConfig.hpp:29]: What a nice and dirty trick!

File: src/cpp/rtps/common/Time_t.cpp [hunk 2]
```diff
@@ -61,6 +61,9 @@ static void current_time_since_unix_epoch(
 namespace eprosima {
 namespace fastrtps {
 
+constexpr int32_t Time_t::INFINITE_SECONDS;
+constexpr uint32_t Time_t::INFINITE_NANOSECONDS;
+
```
Comment (MiguelCompany) [inline src/cpp/rtps/common/Time_t.cpp:66]: Why are these necessary? Are they not obtained from `#include <fastdds/rtps/common/Time_t.h>` ?

File: test/unittest/dds/publisher/PublisherTests.cpp [hunk 3]
```diff
@@ -196,14 +196,213 @@ TEST(PublisherTests, ChangeDefaultDataWriterQos)
...
+    qos.writer_resource_limits().matched_subscriber_allocation.increment = 400;
+    // . data_sharing
+    qos.data_sharing().on("/");
 
     ASSERT_TRUE(publisher->set_default_datawriter_qos(qos) == ReturnCode_t::RETCODE_OK);
     DataWriterQos wqos;
     publisher->get_default_datawriter_qos(wqos);
 
+    // .durability
+    ASSERT_EQ(eprosima::fastdds::dds::TRANSIENT_DURABILITY_QOS, wqos.durability().kind);
```
Comment (MiguelCompany) [inline test/unittest/dds/publisher/PublisherTests.cpp:291]: I would change all the `ASSERT_XXX` into `EXPECT_XXX`

File: test/unittest/dds/subscriber/SubscriberTests.cpp [hunk 4]
```diff
@@ -264,15 +264,235 @@ TEST(SubscriberTests, ChangeDefaultDataReaderQos)
...
+    // .data_sharing
+    qos.data_sharing().on("/");
 
     ASSERT_TRUE(subscriber->set_default_datareader_qos(qos) == ReturnCode_t::RETCODE_OK);
 
     DataReaderQos wqos;
     subscriber->get_default_datareader_qos(wqos);
 
+    // .durability
+    ASSERT_EQ(eprosima::fastdds::dds::TRANSIENT_DURABILITY_QOS, wqos.durability().kind);
```
Comment (MiguelCompany) [inline test/unittest/dds/subscriber/SubscriberTests.cpp:371]: I would change all the `ASSERT_XXX` into `EXPECT_XXX`

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/common/Time_t.cpp (hunk 2 above)
Comment (richiware) [reply src/cpp/rtps/common/Time_t.cpp:66]: Until C++17, `static constexpr` have to be defined, not only declared. In this CPP I am defining the variables.

File: test/unittest/dds/publisher/PublisherTests.cpp (hunk 3 above)
Comment (richiware) [reply test/unittest/dds/publisher/PublisherTests.cpp]: Done

File: test/unittest/dds/subscriber/SubscriberTests.cpp (hunk 4 above)
Comment (richiware) [reply test/unittest/dds/subscriber/SubscriberTests.cpp]: Done

**Review by MiguelCompany — APPROVED** [review]
LGTM with greenish CI

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
There are warnings on Windows CI

**Review by MiguelCompany — DISMISSED** [review]
LGTM with green CI

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

File: test/unittest/dds/publisher/PublisherTests.cpp [hunk 5]
```diff
@@ -196,14 +196,213 @@ TEST(PublisherTests, ChangeDefaultDataWriterQos)
...
+                break;
+            default:
+                EXPECT_TRUE(false);
+        }
+        ++count;
+    }
+    // .ownership
+    EXPECT_EQ(eprosima::fastdds::dds::EXCLUSIVE_OWNERSHIP_QOS, wqos.ownership().kind);
+    // .ownership_strength
+    EXPECT_EQ(30, wqos.ownership_strength().value);
```
Comment (MiguelCompany) [inline test/unittest/dds/publisher/PublisherTests.cpp:356]:
```suggestion
    EXPECT_EQ(30u, wqos.ownership_strength().value);
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
